sycl/doc/extensions/InvokeSIMD/InvokeSIMD.asciidoc

```diff
@@ -0,0 +1,368 @@
+operating on SIMD types as defined there. Additionally, each of the functions
+defined in Sections 4.17.5, 4.17.6 and 4.17.7 of the SYCL 2020 specification
+are overloaded to accept SIMD classes; these overloads operate element-wise on
+the contents of the SIMD class.
+
+[source, c++]
+----
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+namespace simd_abi {
+template <class T, int N>
+using native_fixed_size = /* implementation-defined */;
+} // namespace simd_abi
+
+template <class T, int N>
+using simd = std::experimental::simd<T, simd_abi::native_fixed_size<N>>;
+
+template <class T, int N>
+using simd_mask = std::experimental::simd_mask<T, simd_abi::native_fixed_size<N>>;
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+----
+
+We introduce a new `native_fixed_size` ABI which combines elements of the
+existing `native` and `fixed_size` ABIs defined in Parallelism TS2;
+implementations are expected to use the most efficient representation
+of a `simd<T, N>` type (e.g. using multiple registers rather than an array).
+We expect implementations of this ABI to build on Clang's `ext_vector_type`,
+rather than specifying the register types directly.
+
+The value of `N` for each `simd` argument to a SIMD function must be the same,
+and represents the sub-group size of the calling kernel.  If a function accepts
+`simd` arguments with multiple `N` values, it cannot be invoked from a SPMD
+kernel and the implementation must issue a diagnostic in this case.
+
+The example below shows a simple SIMD function that scales all elements of
+an 8-wide SIMD type `x` by a scalar value `n`.
+
+[source, c++]
+----
+sycl::ext::oneapi::experimental::simd<float, 8> scale(sycl::ext::oneapi::experimental::simd<float, 8> x, float n)
+{
+  return x * n;
+}
+----
+
+=== Invoking a SIMD Function
+
+The `invoke_simd` free function invokes a SIMD function using all work-items
+in a `sub_group`.  The `invoke_simd` interface marshals data between the
+SPMD context of the calling kernel and the SIMD context of the callee,
+converting arguments and return values between scalar and SIMD types as
+appropriate.
+
+[source, c++]
+----
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+template <typename Callable, typename...T>
+auto invoke_simd(sub_group sg, Callable&& f, T... args);
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+----
+
+The `Callable` passed to `invoke_simd` must be a C++ callable that can be
+invoked with the same number of arguments specified in the `args` parameter
+pack. `Callable` may be a function object, a lambda,
+or a function pointer (if the device supports `SPV_INTEL_function_pointers`).
+`Callable` must be an immutable callable with the same type and state for all
+work-items in the sub-group, otherwise behavior is undefined.
```

Comment:
A late nit: I feel like some more explanation would be helpful here: if a callable is a lambda, it is so easy to unintendedly capture something non-uniform, which will then be used inside simd function. Suggest to add non-normative comment that this particular violation of  the `immutable callable with the same type and state for all work-items in the sub-group` principle should be watched out for.

Reply:
Thanks, Konst.  I've added a note in ba8ff0f711becf664b61c6dada597106da44e945.

Reply:
Thanks John!